### PR TITLE
Improve error handling for albums and add support for UtaiteDB and TouhouDB

### PR DESCRIPTION
**Pull request by amogus07**
I recently ran into a problem when running `beet vdbsync`. It would throw the following error:
```
Traceback (most recent call last):
  File "/home/konstantink/.local/bin/beet", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/konstantink/.local/share/pipx/venvs/beets/lib64/python3.12/site-packages/beets/ui/__init__.py", line 1865, in main
    _raw_main(args)
  File "/home/konstantink/.local/share/pipx/venvs/beets/lib64/python3.12/site-packages/beets/ui/__init__.py", line 1852, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/home/konstantink/.local/share/pipx/venvs/beets/lib64/python3.12/site-packages/beetsplug/vocadb.py", line 75, in func
    self.albums(lib, query, move, pretend, write)
  File "/home/konstantink/.local/share/pipx/venvs/beets/lib64/python3.12/site-packages/beetsplug/vocadb.py", line 140, in albums
    item: trackid_to_trackinfo[track_id]
          ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: '3312'
```
It appears that this error occurs when the id of a track of a VocaDB album is changed on the server.
 I had ChatGPT fix it, and it seems to be working.
I also changed log.info to log.debug in some places to make the output of vdbsync more comprehensible.

### CODE REVIEW

**Comment by amogus07** [comment]
If you decide to merge this, please consider updating the description and the tags so it's easier for people to find this plugin.

**Comment by prTopi** [comment]
I've been away from GitHub for a while and haven't had the time to look at this at all, sorry about that.
I'll look this through and maybe comment on some stuff.

But going forward, since I can't quarantee that I will be available on GitHub, do you have any interest in becoming a collaborator for this repo to have push permissions?

File: beetsplug/vocadb.py
Comment (prTopi) [inline beetsplug/vocadb.py:402]: It's a very tiny thing, but I generally go with [EAFP](https://docs.python.org/3/glossary.html#term-EAFP) rather than [LBYL](https://docs.python.org/3/glossary.html#term-LBYL) style, since from what I've heard that's what Python's devs intend people to use.

I'll still accept this change, but if you don't have strong feelings either way, then I'd prefer to stay with EAFP style.

Comment (amogus07) [reply beetsplug/vocadb.py:402]: Didn't know about this, I'll change it back

**Comment by amogus07** [comment]
FYI, I'm currently experimenting with having using an abstract class that `VocaDBPlugin`, `UtaiteDBPlugin` and `TouhouDBPlugin` inherit from, but I'm not sure where to put it for the plugin loader not to detect it and try to load it as a plugin. My current approach is to declare it within a function and have the actual plugins inherit from the return value of that function. You can see my current progress in the dev branch of my fork. Do you have an idea where the abstract class should go? I don't like this approach…

**Comment by prTopi** [comment]
Not quite sure how you would do that in a good way. After looking at beets source for a while, it loads plugin classes from the `beetsplug` namespace, where the classes are subclasses of `BeetsPlugin`.

So 2 ways come to mind for now:
1. the template class wouldn't be an instance of `BeetsPlugin`, and the plugin classes would be instances of `BeetsPlugin` and `VocaDBTemplate`.
2. the template class is placed in another namespace other than `beetsplug`. As in not under the `beetsplug` folder.

I don't like either option, because:
1. I'm pretty sure you can add multiple object instances to a single class (e.g. `VocaDBPlugin(BeetsPlugin, VocaDBTemplate)`) but I've never done that and don't know what's possible with it. Also your IDE won't like the template class not telling what the parent class is.
2. I think this would just add unnecessary "complexity" if we start putting files in multiple locations.

I would like to keep it simple, `vocadb.py` can be the main file (since it's the main project,) and other files import that one.

Though one thing I would change is how the instance is initialized:
```python
# vocadb.py
[...]
class VocaDBPlugin(BeetsPlugin):
    def __init__(self, instance=None) -> None:
        super().__init__()
        if not instance:
            instance = VocaDBInstance(
                name="VocaDB",
                base_url="https://vocadb.net/",
                api_url="https://vocadb.net/api/",
                subcommand="vdbsync",
            )
        self.instance: VocaDBInstance = instance
[...]

# utaitedb.py
from beetsplug.vocadb import VocaDBPlugin, VocaDBInstance

class UtaiteDBPlugin(VocaDBPlugin):
    def __init__(self) -> None:
        super().__init__(
            VocaDBInstance(
                name="UtaiteDB",
                base_url="https://utaiteadb.net/",
                api_url="https://utaitedb.net/api/",
                subcommand="udbsync",
            )
        )
```

**Comment by amogus07** [comment]
Thanks for the feedback! I in the process of implementing the abstract class in another namespace, so I finished that (again, in the dev branch). If you still think it's too complex, I'll just do what you suggested.

**Comment by prTopi** [comment]
It would probably be best to just complete this pull request and open up a new one for the additional changes. I'm ready to merge all of the changes currently here.

Complex was probably not the right word to use by me. I just personally don't really see any upsides in converting the plugin to an abstract class and/or moving it to another namespace.
Then again, I haven't ever used abstract classes so I may just be missing something.

I'm just being a little apprehensive about such changes to the structure of the plugin, as I generally tried to stick close to how other beets plugins have done (this is also why variables like `USER_AGENT` were defined outside of the class.)

Since you've already done work to implement the abstract class changes, if you feel they generally improve the plugin then I am happy to merge them as well.

**Review by prTopi — APPROVED** [review]
LGTM, did some basic tests and seems to work as expected.

I didn't say this anywhere so it is completely optional, but I use `black` as the formatter for the project.
If you want you can format the files using it (If not, then I'll just format them after merging)

Otherwise if the PR is done, mark it as ready and I'll merge

**Comment by amogus07** [comment]
as for the „too complex“ warning for the get_artists method, I‘ll probably move the artist_string logic into it‘s own method

**Comment by amogus07** [comment]
also, black and flake8 are fighting about wether ... should go into it‘s own line in touhoudb.py and utaitedb.py

**Pull request merged**
